plugins/inputs/nfsclient/nfsclient.go

```
@@ -0,0 +1,421 @@
+	"RELEASE_LOCKOWNER",
+	"SECINFO",
+	"FSID_PRESENT",
+	"EXCHANGE_ID",
+	"CREATE_SESSION",
+	"DESTROY_SESSION",
+	"SEQUENCE",
+	"GET_LEASE_TIME",
+	"RECLAIM_COMPLETE",
+	"LAYOUTGET",
+	"GETDEVICEINFO",
+	"LAYOUTCOMMIT",
+	"LAYOUTRETURN",
+	"SECINFO_NO_NAME",
+	"TEST_STATEID",
+	"FREE_STATEID",
+	"GETDEVICELIST",
+	"BIND_CONN_TO_SESSION",
+	"DESTROY_CLIENTID",
+	"SEEK",
+	"ALLOCATE",
+	"DEALLOCATE",
+	"LAYOUTSTATS",
+	"CLONE",
+}
+
+var nfsopFields = []string{
+	"ops",
+	"trans",
+	"timeouts",
+	"bytes_sent",
+	"bytes_recv",
+	"queue_time",
+	"response_time",
+	"total_time",
+	"errors",
+}
+
+func convert(line []string) []int64 {
+	var nline []int64
+	for _, l := range line[1:] {
+		f, _ := strconv.ParseInt(l, 10, 64)
+		nline = append(nline, f)
+	}
+	return nline
+}
+
+func in(list []string, val string) bool {
+	for _, v := range list {
+		if v == val {
+			return true
+		}
+	}
+	return false
+}
+
+func (n *NFSClient) parseStat(mountpoint string, export string, version string, line []string, fullstat bool, nfs3Ops map[string]bool, nfs4Ops map[string]bool, acc telegraf.Accumulator) error {
+	tags := map[string]string{"mountpoint": mountpoint, "serverexport": export}
+	nline := convert(line)
+	first := strings.Replace(line[0], ":", "", 1)
+
+	var fields = make(map[string]interface{})
+
+	if fullstat && first == "events" && len(nline) >= len(eventsFields) {
+		for i, t := range eventsFields {
+			fields[t] = nline[i]
+		}
+		acc.AddFields("nfs_events", fields, tags)
+	} else if fullstat && first == "bytes" && len(nline) >= len(bytesFields) {
+		for i, t := range bytesFields {
+			fields[t] = nline[i]
+		}
+		acc.AddFields("nfs_bytes", fields, tags)
+	} else if fullstat && first == "xprt" && len(line) > 1 {
+		switch line[1] {
+		case "tcp":
+			if len(nline)+2 >= len(xprttcpFields) {
+				for i, t := range xprttcpFields {
+					fields[t] = nline[i+2]
+				}
+				acc.AddFields("nfs_xprt_tcp", fields, tags)
+			}
+		case "udp":
+			if len(nline)+2 >= len(xprtudpFields) {
```

Potential silent dropping here.

---

Technically, we aren't allowed to assume that *anything* in /proc has a stable format. :-(  In fact, there are a few cases in mountstats where I've had to account for slightly different formats between, say kernel 3.x and 5.x.  (See above)

---

It's only about detecting that this plugin has to change. Now you are silently dropping lines if someone in the kernel decides to change the naming to "tcp4" or whatever. Anyway I also agree that this makes the code ugly, so I leave it to you if you want to be on the safe side or not.